Multiple instance Support- Node-DC-EISv1.0

priyankasprakash opened the pull request:
Multi node run allows distributed testing. Multiple applications can be on the same machine or on a pool of machines.

priyankasprakash committed:
Modified post processing of templog files

priyankasprakash committed:
Modified header print statement in runspec.py

priyankasprakash committed:
Changes to server config file to support container runs

priyankasprakash committed:
Added scripts to support multiple instance run
Refactored code to move start and stop server scripts out of server-cluster directory

priyankasprakash committed:
Updating README files
Added a new README file for multiple instance

priyankasprakash committed:
Updating version numbers

priyankasprakash committed:
Updating version in package.json

priyankasprakash committed:
Fixed errors in multiple instance post process file

priyankasprakash committed:
Adding app_host parameter in multiple instance script

priyankasprakash committed:
Added retry for get and post requests

uttampawar commented on Node-DC-EIS-client/README.md:
```diff
@@ -12,9 +12,11 @@ Copyright (c) 2016 Intel Corporation
 
 This client directory contains,
  - runspec.py- A toplevel runspec script; the main script to launch workload.
+     -2 modes: time based run which is the default(mode-1) and request based run (mode-2)
```
Please remove mode-1 and mode-2.

uttampawar commented on README-multiple-instance.MD:
```diff
@@ -0,0 +1,72 @@
+Copyright (c) 2016 Intel Corporation 
```
Convention is to have extension for README document in lower case as, 'README-multiple-instance.md'

uttampawar commented on Node-DC-EIS-client/README.md:
```diff
@@ -12,9 +12,11 @@ Copyright (c) 2016 Intel Corporation
  - runspec.py- A toplevel runspec script; the main script to launch workload.
+     -2 modes: time based run which is the default(mode-1) and request based run (mode-2)
  - node_els-testurls.py - The benchmark driver file which sends actual requests using python's requests module.
  - config.json - The input configuration file.
```
Instead of "input configuration file", we should mention "input client configuration file".

uttampawar commented on container-startup.sh:
```diff
@@ -0,0 +1,41 @@
+export DB_NAME=$5
+export CPU_COUNT=$6
+
+DB_URL=mongodb://${DB_SERVER_IP}:$DB_PORT/$DB_NAME
```
Please add a comment stating assumptions such as, here it's assumed that the mongodb server is already running at the server and port specified in the DB_URL.

uttampawar reviewed (commented):
Please address review comments.

priyankasprakash committed:
Updated client README file
Rename multiple-instance README file

priyankasprakash committed:
Added comments
Changed variable name NODE_SERVER_PORT to SERVER_PORT

egalli commented on Node-DC-EIS-client/process_time_based_output.py:
```diff
@@ -42,23 +42,25 @@ def process_tempfile(results_dir,interval,rampup_rampdown,request,temp_log,insta
     """
+    number_of_files = (2*rampup_rampdown + request)/interval
+    if(((2*rampup_rampdown + request)%interval)!=0):
+       number_of_files +=1
```
these lines could be replace with
```python
number_of_files = int(math.ceil((2.0 * rampup_rampdown + request) / interval))
```
importing math at the top of the file

egalli commented on README-multiple-instance.md:
```diff
@@ -0,0 +1,72 @@
+   	 -2 run modes - bare metal run (default run mode 0) and container run (run mode 1)
+
+ - multiple_instance_config.json - The input configuration file for multiple instances. Each instance requires these parameters:
+ 	  {
```
Do you mind surrounding the JSON block with \`\`\` so it correctly renders in the markdown output.

uttampawar reviewed (approved):
LGTM

egalli commented on README-multiple-instance.md:
```diff
@@ -0,0 +1,72 @@
+
+Multi node run allows distributed testing. You can run multiple applications on the same machine or on a pool of machines. This run typically uses 3 scripts:
+ - run_multiple_instance.sh  - Runs multiple instances of the server client and mongodb on the same machine or across different machines.
+   	 -2 run modes - bare metal run (default run mode 0) and container run (run mode 1)
```
I think you are missing a space in between the - and 2

egalli commented on Node-DC-EIS-client/README.md:
```diff
@@ -12,9 +12,11 @@ Copyright (c) 2016 Intel Corporation
 
 This client directory contains,
  - runspec.py- A toplevel runspec script; the main script to launch workload.
+     -2 modes: time based run which is the default(1) and request based run (2)
```
Missing space between - and 2

egalli commented on README-multiple-instance.md:
```diff
@@ -0,0 +1,72 @@
+Setup instructions:
+Auto-login setup between the client and the server
+    - SSH auto login needs to be set up between the client and the server.RSH and RCP uses this auto login to remote into the server and start the server.
+      -Steps for auto login setup
```
Missing a space between - and 'Steps'

egalli commented on README-multiple-instance.md:
```diff
@@ -0,0 +1,72 @@
+Auto-login setup between the client and the server
+    - SSH auto login needs to be set up between the client and the server.RSH and RCP uses this auto login to remote into the server and start the server.
+      -Steps for auto login setup
+		-$ ssh-keygen -t rsa -b 2048
```
please add \`\`\` around the key gen block

egalli replied:
looks like the markup is still kind of broken. It might be related to spaces vs tab and indentation levels

egalli commented on container-startup.sh:
```diff
@@ -0,0 +1,43 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export SERVER_IP=$1
```
Do you mind adding a check for the correct number of arguments are passed in.

egalli reviewed (commented):
Only some styling things

priyankasprakash committed:
Updating gitignore file for npm-debug.log

priyankasprakash committed:
Updating multiple instance README file

priyankasprakash committed:
Added check for number of arguments in container start-up script

uttampawar reviewed (approved):
LGTM

priyankasprakash committed:
Calculating number of file using math ceil function

priyankasprakash committed:
Formatting README file

egalli committed:
Fixing markdown output

Markdown seems to be fairly picky when it comes to indentation. Specially when placing code blocks inside lists.

egalli commented:
LGTM2

uttampawar committed:
Update README-multiple-instance.md

uttampawar reviewed (approved):
LGTM
